### PR TITLE
add database seeder to backend (#64)

### PR DESCRIPTION
**Pull request by mhalsubhi**
Added database seeder to backend 

local database in local development environment now can be seeded with actual data fetched from [api](https://finalspaceapi.com/api/v0/)

to seed database after clone the repo and cd to backend folder, use command `npm run seed` to seed local database

#64

### CODE REVIEW

**Comment by lelouchB** [comment]
Hey @mhalsubhi 
This looks awesome 👌 
I will get to it sometime tommorow.

I am going to create an issue to properly demonstrate this in `backend/README.md`

Are you interested in that.

**Comment by mhalsubhi** [comment]
@lelouchB yes, I'll create step by step guide for it 👍🏼

**Comment by lelouchB** [comment]
> @lelouchB yes, I'll create step by step guide for it 👍🏼 

Comment on #92 so I can assign it you.

**Comment by lelouchB** [comment]
> This pull request **introduces 2 alerts** when merging [1ac153f](https://github.com/lelouchB/final-space-api/commit/1ac153ff388e9756a5aece3fb7f4c863f834d090) into [f59d552](https://github.com/lelouchB/final-space-api/commit/f59d5521e8b75f113f837f23cc18f5b3fa9f220c) - [view on LGTM.com](https://lgtm.com/projects/g/lelouchB/final-space-api/rev/pr-51757e05c7c1349ce4c1e178723cf8ec69d2c984)
> 
> **new alerts:**
> 
> * 2 for Missing variable declaration

Can you look into this

**Comment by mhalsubhi** [comment]
@lelouchB maybe if changed it to `for (var character of characters) ` it will be solved it, I'll try now 👍🏼

**Comment by mhalsubhi** [comment]
@lelouchB resolved 👍🏼

File: backend/seeder.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      return false
+    })
+}
+
+closeDb = async () => {
+  mongoose.connection.close()
+}
+
+seedCharacters = async () => {
+  let charactersEndpoint = "https://finalspaceapi.com/api/v0/character"
```
Comment (JuanVqz) [inline backend/seeder.js:36]: what about to store the BASE_URL_API=https://finalspaceapi.com/api on the .env.example
```suggestion
  let charactersEndpoint = "${BASE_URL_API}/v0/character"
```

Comment (lelouchB) [reply backend/seeder.js:36]: I guess we need another environment variable  something related to PORT for this.
Create a new environment variable and use it instead here.

Comment (JuanVqz) [reply backend/seeder.js:36]: that is which I wanted to say, hahaha

**Comment by lelouchB** [comment]
@mhalsubhi 
I tested this on my local machine, it works great.
Nice 👍 
Can you move `seeder.js` to another directory. It should not be in root.
Maybe `helpers` or a new directory `seeder`?

Thanks

**Comment by mhalsubhi** [comment]
@lelouchB sure, I will move the file to `helpers` folder

**Comment by lelouchB** [comment]
Hey @mhalsubhi 
Can you resolve these conflicts so I can merge this?

Thanks

**Comment by lelouchB** [comment]
@mhalsubhi 
I am going to remove the images from README.md. They just don't fulfill the purpose I wanted. The problem is not with the images, I am sorry for doing this after making you add them.

I apologize for this.

**Comment by mhalsubhi** [comment]
> Hey @mhalsubhi
> Can you resolve these conflicts so I can merge this?
> 
> Thanks

@lelouchB I will try 👍🏼  

> @mhalsubhi
> I am going to remove the images from README.md. They just don't fulfill the purpose I wanted. The problem is not with the images, I am sorry for doing this after making you add them.
> 
> I apologize for this.

No worries it's totally ok for me 😃

**Review by lelouchB — APPROVED** [review]

**Comment by lelouchB** [comment]
hey @mhalsubhi 
there are some linting issues do you think you can send a fix for these?
Don't why but  `npm run lint:fix` command is not working properly

**Comment by mhalsubhi** [comment]
@lelouchB sure, I will give it a try 👍🏼

**Comment by lelouchB** [comment]
> hey @mhalsubhi 
> there are some linting issues do you think you can send a fix for these?
> Don't why but  `npm run lint:fix` command is not working properly

Maybe the errors are not auto fixable.

**Pull request merged**